### PR TITLE
Shelf modal 665

### LINKED ISSUE #665: Create add shelf modal
**Describe the solution you'd like**

image

Create an add shelf modal that is centered in the middle of the screen. It should also be responsive.

The modal should be brought up when the user clicks on 'Add shelf'. When the user clicks on the 'cancel' button in the modal, it should close.

### PR DESCRIPTION
**Pull request by olusolaa**
## Summary of change

 Created The Add Shelf Model

## Related issue

Closes #665

## Additional context

**Desktop View**
Screen Shot 2021-05-17 at 2 44 08 AM

**IPad view**
Screen Shot 2021-05-17 at 2 49 00 AM

**Mobile view**
Screen Shot 2021-05-17 at 2 45 24 AM

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests 

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [x] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [x] Created a branch that has a descriptive name (what your branch is for in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

For any of the optional checkboxes (e.g. the screenshots one), still check it if it does not apply.

### CODE REVIEW

**Comment by knjk04** [comment]
Thanks for working on this! Could you please resolve the merge conflict?

**Comment by olusolaa** [comment]
> Thanks for working on this! Could you please resolve the merge conflict?

I have resolved the conflict.

File: frontend/src/my-books/myBookModal/ShelfModal.css
Comment (RivaD2) [inline frontend/src/my-books/myBookModal/ShelfModal.css:71]: We can remove lines 71 and 75 (if these are unused styles)

File: frontend/src/my-books/myBookModal/ShelfModal.css
Comment (RivaD2) [inline frontend/src/my-books/myBookModal/ShelfModal.css:14]: This is looking really great! Let's bump up the font-size for the .modal-title to 40pt. Double-check line spacing here as well and indent for new line on 46 and 57.

File: frontend/src/my-books/myBookModal/ShelfModal.css
Comment (RivaD2) [inline frontend/src/my-books/myBookModal/ShelfModal.css:20]: Let's bump up the font-size to 16pt for .shelf-modal-desc-container

File: frontend/src/my-books/MyBooks.css
Comment (RivaD2) [inline frontend/src/my-books/MyBooks.css:19]: For this file, line-spacing just needs to be set to 4.

File: frontend/src/my-books/myBookModal/ShelfModal.css
Comment (RivaD2) [inline frontend/src/my-books/myBookModal/ShelfModal.css:73]: Let's add some space in between the /shelf-modal-buttons for mobile. Right now they are displayed as a column in mobile which great! We just need to push them apart about 24-26px(let's try that).

File: frontend/src/my-books/MyBooks.tsx
Comment (RivaD2) [inline frontend/src/my-books/MyBooks.tsx:21]: Delete whitespace here (line 21)

File: frontend/src/my-books/MyBooks.tsx
Comment (RivaD2) [inline frontend/src/my-books/MyBooks.tsx:61]: clear whitespace on line 60-61

File: frontend/src/my-books/MyBooks.tsx
Comment (RivaD2) [inline frontend/src/my-books/MyBooks.tsx:96]: Here, line 63 is a child of line 62, so we can indent line 63 accordingly and edit indentation for this entire block, No whitespace is needed for lines 80, 84, 96.

File: frontend/src/my-books/myBookModal/Modal.tsx
Comment (RivaD2) [inline frontend/src/my-books/myBookModal/Modal.tsx:43]: We can remove any unused styles here and just double check your line spacing for the file again :)

File: frontend/src/my-books/myBookModal/ShelfModal.tsx
Comment (RivaD2) [inline frontend/src/my-books/myBookModal/ShelfModal.tsx:85]: Starting at line 39, we will want to clean up indentation and whitespace. Starting on line 39, indent accordingly for the block. If you have questions I am more than happy to help :)

**Review by RivaD2 — COMMENTED** [review]
Really awesome work here!
   I left comments regarding line-spacing and some edits needed for font-size and spacing for buttons in mobile view. 

   It looks like you have created another Modal using Material-UI. To mitigate redundant code, you can instead use our 
   responsive modal that is already built (it is in the `components` folder). Import in our `Modal.tsx` file instead into `ShelfModal.tsx`, and then migrate your styles accordingly from your newly created`Modal.css` file. 

I happy to help if you have any questions!

File: frontend/src/my-books/myBookModal/ShelfModal.tsx
Comment (RivaD2) [inline frontend/src/my-books/myBookModal/ShelfModal.tsx:19]: Import in our Modal here instead (found in the `components` folder)

**Comment by olusolaa** [comment]
@RivaD2 Thanks for the review, I hope the recent commits I made, address the corrections. Please let me know if there are other things to adjust.

File: frontend/src/my-books/myBookModal/ShelfModal.tsx
Comment (olusolaa) [reply frontend/src/my-books/myBookModal/ShelfModal.tsx:19]: The modal in the components folder has a fixed width of 380px which looks too big for a smaller screen. I had to change the width to 50%. This has affected the width of the change password modal. Please let me know if there are better approaches to take.

Comment (RivaD2) [reply frontend/src/my-books/myBookModal/ShelfModal.tsx:19]: The 380 set is not for mobile design but rather for the base Modal itself on desktop. To get these modals to be responsive, we need to use Material-UI's `useMediaQuery`hook. I have successfully got this working on my end so all modals are now responsive. 
  For now, you can leave your code as is and I will go back into my branch (after we get your branch reviewed and merged) to make necessary changes to the base Modal.

Comment (RivaD2) [reply frontend/src/my-books/myBookModal/ShelfModal.tsx:19]: I have started your review and will get back in touch with you soon :)

Comment (RivaD2) [reply frontend/src/my-books/myBookModal/ShelfModal.tsx:19]: @olusolaa Hello! I added the fix for the modal in my PR for another ticket. We will work on getting that merged first so you can then take in those changes. Hopefully once that happens, the modal will be responsive on your end and you can double-check that the rest of your responsive styles still work.

File: frontend/src/my-books/ShelfModal.css
Comment (RivaD2) [inline frontend/src/my-books/ShelfModal.css:56]: We can't guarantee that Material-UI will always assign 46 to the element being targeted here. Please create a unique class specific to this element. The same applies to other styles in this file.

File: frontend/src/my-books/ShelfModal.tsx
Comment (RivaD2) [inline frontend/src/my-books/ShelfModal.tsx:4]: Change year to 2021

File: frontend/src/my-books/ShelfModal.tsx
Comment (RivaD2) [inline frontend/src/my-books/ShelfModal.tsx:39]: Please remove the alert. If there is a ticket to hook this up to a backend, please leave a comment with the ticket number.

File: frontend/src/my-books/ShelfModal.tsx
Comment (RivaD2) [inline frontend/src/my-books/ShelfModal.tsx:92]: You're buttons are displaying correctly for desktop, when they are flexed in a row. However, for mobile, the Cancel button should be below Add Shelf button. In media query, you can use css flex styles to reverse the order of the children. Additionally, please add space between the buttons.

File: frontend/src/shared/components/Modal.tsx
Comment (RivaD2) [inline frontend/src/shared/components/Modal.tsx:17]: Please revert all changes to this file. I have made edits in a separate ticket to address the responsive issues :)

**Review by RivaD2 — COMMENTED** [review]
Thank you for making the changes, I added a few more notes primarily to make styles more robust. 
Review your files once more and change all spacing to 4. 
Please delete the package-lock.json. Then commit that deletion.

File: frontend/src/my-books/ShelfModal.tsx
Comment (olusolaa) [reply frontend/src/my-books/ShelfModal.tsx:92]: In the last commit, I already added 24px to push the button apart in the mobile view, do you mean I should increase the space to about 30px?

Comment (RivaD2) [reply frontend/src/my-books/ShelfModal.tsx:92]: Can you update your screenshot so we can see how the 24 px looks? Thank you :)

**Comment by olusolaa** [comment]
How do I set line spacing to 4 for all files?

**Comment by RivaD2** [comment]
> How do I set line spacing to 4 for all files?

@olusolaa If you're using VS code as your code editor, at the bottom of each file you will see the "Spaces" section. Yours will most likely say, "Spaces: 2". Click on "Spaces", then "Indent Using Spaces", and then change spaces to 4. You can also go to 'Code' -> "Preferences" -> "Settings" to edit spacing.

We will soon have a linter to help us catch style violations (spacing errors etc) so this will be easier. In the meantime, before pushing our contributions, we will need to check that our spaces are set to 4 as the default spacing.

If you are not using VS code, let me know and I will be happy to help!

**Review by RivaD2 — APPROVED** [review]
Thank you for making the changes ( we are almost there :) )

**Review by RivaD2 — APPROVED** [review]

**Comment by olusolaa** [comment]
image

File: frontend/src/my-books/myBookModal/ShelfModal.css
Comment (RivaD2) [reply frontend/src/my-books/myBookModal/ShelfModal.css]: For this, let's take it down to 14pt and we should be good :)

File: frontend/src/my-books/myBookModal/ShelfModal.css
Comment (RivaD2) [reply frontend/src/my-books/myBookModal/ShelfModal.css]: Alrighty, now that I see the 40px, it is a bit too large. Let's take it down closer to where you had it, at 22px. We should be good then.

**Comment by RivaD2** [comment]
@olusolaa Thank you for sending the screenshot above. The button spacing looks good. Can you tell me what the hamburger menu is in the background?

**Comment by olusolaa** [comment]
The title and the two buttons (add book and add shelf button) at the top collapse on mobile to the menu bar. It not part of the issue assigned to me though.

**Comment by RivaD2** [comment]
@olusolaa Please bring your branch even with 0.2.0. I left a comment regarding `.modal-title`. Let's take the font-size back down to `22pt`. 

Let me know if you have any questions 👍

**Comment by RivaD2** [comment]
@knjk04 Hi there! 

Can you change the branch in this PR to `https://github.com/RivaD2/book-project/tree/shelf-modal-665`. I suppose we can also delete this PR now and I can just open a new one. Let me know what works best.

I don't have permission to edit PR's (perhaps you can give me access) 👍

**Comment by olusolaa** [comment]
> @olusolaa Please bring your branch even with 0.2.0. I left a comment regarding `.modal-title`. Let's take the font-size back down to `22pt`.
> 
> Let me know if you have any questions 👍

Sorry for the late response, I relocated and just finding my balance. Should I still go ahead with this?

**Comment by RivaD2** [comment]
> > @olusolaa Please bring your branch even with 0.2.0. I left a comment regarding `.modal-title`. Let's take the font-size back down to `22pt`.
> > Let me know if you have any questions 👍
> 
> Sorry for the late response, should I still go ahead with this?

Yes, that we be great :) 👍 
After pulling down your branch, I can see a couple of places we need to make some small fixes. I will add that review shortly.

File: frontend/src/my-books/MyBooks.css
Comment (RivaD2) [inline frontend/src/my-books/MyBooks.css:23]: Remove the lines

File: frontend/src/my-books/MyBooks.css
Comment (RivaD2) [inline frontend/src/my-books/MyBooks.css:31]: Please delete line

File: frontend/src/my-books/MyBooks.tsx
Comment (RivaD2) [inline frontend/src/my-books/MyBooks.tsx:54]: Bring back in `Layout` component. Replace this div with `<Layout title="My books">` and close the  Layout tag 
at bottom.

File: frontend/src/my-books/MyBooks.tsx
Comment (RivaD2) [inline frontend/src/my-books/MyBooks.tsx:56]: Remove Hidden tag here. The buttons should show on mobile. We don't yet have responsive designs so for now, we can just have them showing.

File: frontend/src/my-books/MyBooks.tsx
Comment (RivaD2) [inline frontend/src/my-books/MyBooks.tsx:79]: Remove these lines

File: frontend/src/my-books/ShelfModal.css
Comment (RivaD2) [inline frontend/src/my-books/ShelfModal.css:87]: Remove !important as above comment fixes this.

File: frontend/src/my-books/ShelfModal.css
Comment (RivaD2) [inline frontend/src/my-books/ShelfModal.css:83]: Add in :    
  `.shelf-button-container > .MuiButton-containedPrimary {
        margin-bottom:14px;
    } `

File: frontend/src/my-books/ShelfModal.css
Comment (RivaD2) [inline frontend/src/my-books/ShelfModal.css:73]: Please Delete lines as these styles are redundant (they are set in your base styles) 👍

File: frontend/src/my-books/ShelfModal.css
Comment (RivaD2) [inline frontend/src/my-books/ShelfModal.css:75]: Please change `max-width` in query to `600px`

File: frontend/src/my-books/ShelfModal.css
Comment (RivaD2) [inline frontend/src/my-books/ShelfModal.css:51]: Please add in new property of `whitespace:nowrap`

**Review by RivaD2 — CHANGES_REQUESTED** [review]
Really nice work!

  Normally I wouldn't ask you to add additional styles in that aren't related to your ticket, but there are two more fixes to help with your modal sizing (as well as keep things uniform in all modals from now on).  In addition to the requested changes below, in:

`ForgotPasswordModal.tsx` : change media query to be `max-Width:600px`
`Modal.tsx`: Change the following in Material-UI 's `createStyles`:

       `maxWidth: '80%',
      [theme.breakpoints.between(300, 600)]:{maxWidth: '60%'},`

Let me know if I can help with anything 👍

File: frontend/src/my-books/ShelfModal.css
Comment (RivaD2) [reply frontend/src/my-books/ShelfModal.css]: Please remove entire line, no need as the below comment will set the margin on the Material-UI element :)

File: frontend/src/my-books/ShelfModal.css
Comment (RivaD2) [reply frontend/src/my-books/ShelfModal.css]: resolved, hiding comment.

File: frontend/src/my-books/ShelfModal.css
Comment (RivaD2) [inline frontend/src/my-books/ShelfModal.css]: It looks like you only removed the comment here, please remove line 87 as well. 👍

Comment (RivaD2) [reply frontend/src/my-books/ShelfModal.css]: My bad, you got this one! 👍  Please ignore :)

**Comment by RivaD2** [comment]
@olusolaa  looks like all that is left is to remove the entire margin property from `.shelf-modal-button' and get your branch even with 0.2.0. Then we are ready to merge 👍 💯

**Review by RivaD2 — CHANGES_REQUESTED** [review]
Hello there @olusolaa,

It looks like you meant to delete the node modules file, but instead deleted the frontend/yarn.lock and the yarn.error.log. I can see you ran  `npm -i`  from the repo root, not the frontend folder :)

Please:
 - revert your previous two commits (as we need to get your branch back to the state it was in before node modules was committed).
 -  delete the node modules file and make a commit right after
 - Then you can make a new commit that adds the node modules to the .`gitignore.` If the node modules is in the `.gitignore` first, then deleting them won't commit the deletion and they will still be there. Once that is done, commit any further work separately.

 I am happy to help should you have questions 👍

**Comment by olusolaa** [comment]
@RivaD2, I'm yet to get a response from you about my latest commit. Is there something else I need to fix?

**Comment by RivaD2** [comment]
> @RivaD2, I'm yet to receive a response from you about my last commit. Please let me know if there are further changes to make. I will be glad to carry them out.

@olusolaa If you take a look at the `files Changed` section at the top of this PR, you will see over 2800 changes. Most of these are from the node modules folder that needs to be deleted. When this PR is ready, it should have maybe a dozen changes. When you are all done, please merge latest 0.2.0 into this branch and let me know. Thanks for sticking with it 👍

**Comment by olusolaa** [comment]
> > @RivaD2, I'm yet to receive a response from you about my last commit. Please let me know if there are further changes to make. I will be glad to carry them out.
> 
> @olusolaa If you take a look at the `files Changed` section at the top of this PR, you will see over 2800 changes. Most of these are from the node modules folder that needs to be deleted. When this PR is ready, it should have maybe a dozen changes. When you are all done, please merge latest 0.2.0 into this branch and let me know. Thanks for sticking with it 👍

Thanks, @RivaD2. I have made the required changes.

**Review by RivaD2 — APPROVED** [review]
Thank you for the updates, this looks good! 👍

**Review by knjk04 — CHANGES_REQUESTED** [review]
@olusolaa Thanks for working on this and for making the changes suggested!

image

It looks like we're displaying 'My books' twice. Also, the buttons look quite strange. See the issue description for how we want them to look. Would it be possible to address these points?

**Comment by olusolaa** [comment]
> It looks like we're displaying 'My books' twice. Also, the buttons look quite strange. See the issue description for how we want them to look. Would it be possible to address these points?

Sure @knjk04, this is how the layout appears now.
image

**Review by knjk04 — APPROVED** [review]
Thanks, this looks great!

**Pull request merged**
